datafusion/core/src/execution/context.rs

```diff
@@ -1663,783 +1655,48 @@ mod tests {
-                file.write_all(data.as_bytes())?;
-            }
-        }
+        let variable_provider = test::variable::SystemVar::new();
+        ctx.register_variable(VarType::System, Arc::new(variable_provider));
+        let variable_provider = test::variable::UserDefinedVar::new();
+        ctx.register_variable(VarType::UserDefined, Arc::new(variable_provider));
 
-        ctx.register_csv(
-            "test",
-            tmp_dir.path().to_str().unwrap(),
-            CsvReadOptions::new().schema(&schema).has_header(false),
-        )
-        .await?;
+        let provider = test::create_table_dual();
+        ctx.register_table("dual", provider)?;
 
-        let results = plan_and_collect(
-            &ctx,
-            "SELECT date_trunc('week', t1) as week, SUM(c2) FROM test GROUP BY date_trunc('week', t1)",
-        ).await?;
+        let results =
+            plan_and_collect(&ctx, "SELECT @@version, @name, @integer + 1 FROM dual")
+                .await?;
 
         let expected = vec![
-            "+---------------------+--------------+",
-            "| week                | SUM(test.c2) |",
-            "+---------------------+--------------+",
-            "| 2020-12-07 00:00:00 | 24           |",
-            "| 2020-12-14 00:00:00 | 156          |",
-            "+---------------------+--------------+",
+            "+----------------------+------------------------+------------------------+",
+            "| @@version            | @name                  | @integer Plus Int64(1) |",
+            "+----------------------+------------------------+------------------------+",
+            "| system-var-@@version | user-defined-var-@name | 42                     |",
+            "+----------------------+------------------------+------------------------+",
         ];
-        assert_batches_sorted_eq!(expected, &results);
+        assert_batches_eq!(expected, &results);
 
         Ok(())
     }
 
     #[tokio::test]
-    async fn group_by_largeutf8() {
-        {
-            let ctx = SessionContext::new();
-
-            // input data looks like:
-            // A, 1
-            // B, 2
-            // A, 2
-            // A, 4
-            // C, 1
-            // A, 1
-
-            let str_array: LargeStringArray = vec!["A", "B", "A", "A", "C", "A"]
-                .into_iter()
-                .map(Some)
-                .collect();
-            let str_array = Arc::new(str_array);
-
-            let val_array: Int64Array = vec![1, 2, 2, 4, 1, 1].into();
-            let val_array = Arc::new(val_array);
-
-            let schema = Arc::new(Schema::new(vec![
-                Field::new("str", str_array.data_type().clone(), false),
-                Field::new("val", val_array.data_type().clone(), false),
-            ]));
-
-            let batch =
-                RecordBatch::try_new(schema.clone(), vec![str_array, val_array]).unwrap();
+    async fn register_deregister() -> Result<()> {
```

Comment:
👍  for keeping the variable tests (and others that are related to `SessionContext` in the same file